### PR TITLE
Expire deprecations from 3.6

### PR DESCRIPTION
**Pull request by oscargus**
## PR Summary

Not all, but most (all?) of the ones I understood (quite) easily...

Split into multiple commits for somewhat sensible commit messages. Will add an api-change note.

The first commit also improves the argument checking of `(set_)pickradius` slightly.

## PR Checklist

**Documentation and Tests**
- [ ] Has pytest style unit tests (and `pytest` passes)
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] New plotting related features are documented with examples.

**Release Notes**
- [ ] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [ ] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [ ] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

File: doc/users/prev_whats_new/dflt_style_changes.rst [hunk 1]
```diff
@@ -575,7 +575,7 @@ default.  The default face color is now ``'C0'`` instead of ``'b'``.
            ax_bottom.set_ylim(0, .75)
            ax_bottom.add_artist(mpatches.Rectangle(grid[1] - [0.025, 0.05],
                                                    0.05, 0.1))
-           ax_bottom.add_artist(mpatches.RegularPolygon(grid[3], 5, 0.1))
```
Comment (oscargus) [inline doc/users/prev_whats_new/dflt_style_changes.rst:578]: One would have expected this to warn earlier?

**Comment by efiring** [comment]
Thank you for doing this.  I read through the file diffs without finding any problems.  I did not try to match every removal with a note in the changes/removals file, but I did notice that removals from the backends don't seem to be mentioned there.

**Comment by oscargus** [comment]
Ahh, you are correct. Well spotted!

In the 3.6 documents it was a big list of all the backend changes, but I didn't touch the interactive ones, so at the time I just skipped that for later and forgot about it...

**Comment by oscargus** [comment]
This is now added. 

Note that some of the things didn't have deprecation notices in the 3.6 docs, such as positional argument renaming.

File: doc/api/next_api_changes/removals/24984-OG.rst [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+replaced by ``set_base`` and ``set_label_minor``, respectively.
+
+In `~.LogLocator`, the methods ``base`` and ``subs`` for setting the respective
+parameter are removed. Instead, use ``set_params(base=..., subs=...)``.
+
+``Axes.get_renderer_cache``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The canvas now takes care of the renderer and whether to cache it or not,
+so the ````Axes.get_renderer_cache`` method is removed. The
```
Comment (anntzer) [inline doc/api/next_api_changes/removals/24984-OG.rst:44]: quadruple backquote

File: lib/matplotlib/mlab.py [hunk 3]
```diff
@@ -233,8 +229,6 @@ def stride_windows(x, n, noverlap=None, axis=0):
         The number of data points in each window.
     noverlap : int, default: 0 (no overlap)
         The overlap between adjacent windows.
-    axis : int
-        The axis along which the windows will run.
 
     References
     ----------
```
Comment (anntzer) [inline lib/matplotlib/mlab.py:234]: You can probably at least remove the SO links (which predate sliding_window_view); FWIW I would just get rid of the entire function and inline it into _spectral_helper (and this way you only need to check once that noverlap < NFFT).

File: lib/mpl_toolkits/mplot3d/axis3d.py [hunk 4]
```diff
@@ -198,11 +183,6 @@ def get_minor_ticks(self, numticks=None):
                 obj.set_transform(self.axes.transData)
         return ticks
 
-    @_api.deprecated("3.6")
-    def set_pane_pos(self, xys):
-        """Set pane position."""
-        self._set_pane_pos(xys)
-
     def _set_pane_pos(self, xys):
```
Comment (anntzer) [inline lib/mpl_toolkits/mplot3d/axis3d.py:186]: You can inline _set_pane_pos to its only call site too (as I did at https://github.com/matplotlib/matplotlib/pull/25373/files#diff-ea1dd4bc7c3b7c0078e69e41e916d25354864e4f2fe31fd71210477f6f9f9660R310; note in particular that the staleness-setting in _set_pane_pos is redundant as we should be in the middle of a draw and thus clearing staleness right now.

**Comment by oscargus** [comment]
Thanks @anntzer ! Should be sorted out now.

**Review by anntzer — APPROVED** [review]
postci

File: lib/matplotlib/collections.py [hunk 5]
```diff
@@ -443,6 +439,9 @@ def set_pickradius(self, pickradius):
         pickradius : float
             Pick radius, in points.
         """
+        if not isinstance(pickradius, Number):
```
Comment (ksunden) [inline lib/matplotlib/collections.py:442]: Do we want this check to be for `Real` instead of `Number`?

Comment (ksunden) [reply lib/matplotlib/collections.py:442]: and should this also include a non-negative check like pickradius for Line2D?

File: lib/matplotlib/offsetbox.py [hunk 6]
```diff
@@ -1229,8 +1225,7 @@ def __str__(self):
         return f"AnnotationBbox({self.xy[0]:g},{self.xy[1]:g})"
 
     @_docstring.dedent_interpd
-    @_api.make_keyword_only("3.6", name="xycoords")
-    def __init__(self, offsetbox, xy,
+    def __init__(self, offsetbox, xy, *,
```
Comment (ksunden) [inline lib/matplotlib/offsetbox.py:1228]: `*` should be after `xybox`, as `xycoords` was the indicated parameter

**Review by ksunden — CHANGES_REQUESTED** [review]
Only hard blocker is the offsetbox kw-only arg position, but also commented on collections version of `pickradius` if you wish to make that consistent with some other pickradus implementations

**Comment by oscargus** [comment]
Thanks @ksunden ! Good finds!

**Pull request merged**